SPECS/tini/tini.spec

```diff
@@ -1,30 +1,49 @@
 Summary:        A tiny but valid init for containers
 Name:           tini
 Version:        0.19.0
-Release:        6%{?dist}
+Release:        7%{?dist}
 License:        MIT
 Vendor:         Microsoft Corporation
 Distribution:   Mariner
 URL:            https://github.com/krallin/tini
 Source0:        https://github.com/krallin/tini/archive/v%{version}.tar.gz#/%{name}-%{version}.tar.gz
+BuildRequires:  binutils
 BuildRequires:  cmake
+BuildRequires:  diffutils
+BuildRequires:  file
 BuildRequires:  gcc
-BuildRequires:  glibc-devel
+BuildRequires:  glibc-devel >= 2.34-3
+BuildRequires:  kernel-headers
+BuildRequires:  make
 BuildRequires:  sed
 
-Provides:       docker-init = %{version}-%{release}
-
 %description
 Tini is the simplest init you could think of.
 
 All Tini does is spawn a single child (Tini is meant to be run in a container),
 and wait for it to exit all the while reaping zombies and performing signal
 forwarding.
 
+%package static
+Summary:        Standalone static build of tini
+# `docker-init` used to be provided by `tini` it's now provided by `tini-static`
+# `tini` and `tini-static` are co-installable so long as both are newer than
+# that change.
+Conflicts:      %{name} <= 0.19.0-6
+Provides:       docker-init = %{version}-%{release}
+
+%description static
+This package contains a standalone static build of tini, meant to be used
+inside a container.
+
 %prep
 %autosetup
 # Do not strip binaries
 sed -i CMakeLists.txt -e 's/ -Wl,-s//'
+# Enable static-pie (ASLR) support for tini-static
+sed -i CMakeLists.txt -e 's/ -static/ -static-pie/'
+# Enable RELRO NOW for all binaries
+sed -i CMakeLists.txt -e 's/ -Wl,-z,relro/ -Wl,-z,relro,-z,now/'
```

Comment:
Both `-z,relro` and `-z,now` are part of Mariner's default `LDFLAGS` (unless you add `%undefine _hardened_build` to the spec but that's not the case here). Is `tini` ignoring the defaults? It might be a better solution to have it build with the defaults, so we're sure others security hardenings are applied as well.

Reply:
I did notice that, but only after I'd made this change 😄.  I've just retested and yes, the default flags are indeed passed (and this change was thus redundant).